charmcraft/shell.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Shell-related utilties."""
+
+import logging
+import pathlib
+import subprocess
+from typing import Optional
+
+logger = logging.getLogger(__name__)
+
+
+def launch_shell(*, cwd: Optional[pathlib.Path] = None) -> None:
+    """Launch a user shell for debugging environment.
+
+    :param cwd: Working directory to start user in.
+    """
+    subprocess.run(["bash"], check=False, cwd=cwd)
```

Comment:
Does it worth to have a whole new module, only to have inside ONE function, that even has ONE line?

Do we plan for this module to grow in the future?

Reply:
I placed it in its own module to prevent unnecessary coupling between different modules I expected to use it.  I had originally expected that `parts` module would also use it, but that's no longer the case.  OTOH, I expect it to get a little more complex in a cross-OS future.

I can consolidate it into `commands/build` and import from `commands/pack` ?